SingularityService/src/main/java/com/hubspot/singularity/data/InactiveSlaveManager.java

```diff
@@ -0,0 +1,53 @@
+package com.hubspot.singularity.data;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.curator.framework.CuratorFramework;
+
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import com.hubspot.singularity.config.SingularityConfiguration;
+import com.hubspot.singularity.data.transcoders.Transcoder;
+
+@Singleton
+public class InactiveSlaveManager extends CuratorManager {
+  private static final String ROOT_PATH = "/inactive";
+  private final Transcoder<String[]> inactiveSlaveTranscoder;
+
+  @Inject
+  public InactiveSlaveManager(CuratorFramework curator,
+                              SingularityConfiguration configuration,
+                              MetricRegistry metricRegistry,
+                              Transcoder<String[]> inactiveSlaveTranscoder) {
+    super(curator, configuration, metricRegistry);
+
+    this.inactiveSlaveTranscoder = inactiveSlaveTranscoder;
+  }
+
+  public Set<String> getInactiveSlaves() {
+    Optional<String[]> maybeInactiveSlaves = getData(ROOT_PATH, this.inactiveSlaveTranscoder);
```

Comment:
Wondering if it might be easier to just create empty child nodes with the proper name here, rather than continually reading/writing an array. Creation would just be creating a node at `/inactive/(hostname)`, getting all would be a `getChildren()` call, and checking if an individual one would just be an `exists()` call rather than grabbing the whole array.